docs/using-hive.md

```diff
@@ -77,87 +236,125 @@ metadata:
-    serviceCIDR: 172.30.0.0/16
-    type: OpenShiftSDN
+        type: m4.xlarge
+```
+
+
+An example `ClusterDeployment` for AWS:
+
+```yaml
+apiVersion: hive.openshift.io/v1
+kind: ClusterDeployment
+metadata:
+  name: mycluster
+spec:
+  baseDomain: hive.example.com
+  clusterName: mycluster
+  controlPlaneConfig:
+    servingCertificates: {}
+  installed: false
   platform:
     aws:
-      region: us-east-1
-  platformSecrets:
-    aws:
-      credentials:
+      credentialsSecretRef:
         name: mycluster-aws-creds
-  pullSecret:
+      region: us-east-1
+  provisioning:
+    imageSetRef:
+      name: openshift-v4.3.0
+    installConfigSecretRef:
+      name: mycluster-install-config
+    sshPrivateKeySecretRef:
+      name: mycluster-ssh-key
+  pullSecretRef:
     name: mycluster-pull-secret
-  sshKey:
-    name: mycluster-ssh-key
 ```
 
-The hiveutil CLI (see `make hiveutil`) offers a create-cluster command for generating a cluster deployment and submitting it to the Hive cluster using your current kubeconfig.
-
-To view what create-cluster generates, *without* submitting it to the API server, add `-o yaml` to the above command. If you need to make any changes not supported by create-cluster options, the output can be saved, edited, and then submitted with `oc apply`.
-
-By default this command assumes the latest Hive master CI build, and the latest OpenShift stable release. `--release-image` can be specified to control which OpenShift release image to install in the cluster.
-
-#### Create Cluster on AWS
-
-Credentials will be read from your AWS_ACCESS_KEY_ID and AWS_SECRET_ACCESS_KEY environment variables. Alternatively you can specify an AWS credentials file with --creds-file.
+For Azure, replace `spec.platform` with:
 
-```bash
-bin/hiveutil create-cluster --base-domain=mydomain.example.com --cloud=aws mycluster
+```yaml
+azure:
+  baseDomainResourceGroupName: my-bdrgn
+  credentialsSecretRef:
+    name: mycluster-azure-creds
+  region: centralus
 ```
 
-#### Create Cluster on Azure
+For GCP, replace `spec.platform` with:
 
-Credentials will be read from `~/.azure/osServicePrincipal.json` typically created via the `az login` command.
-
-```bash
-bin/hiveutil create-cluster --base-domain=mydomain.example.com --cloud=azure --azure-base-domain-resource-group-name=myresourcegroup mycluster
+```yaml
+gcp:
+  credentialsSecretRef:
+    name: mycluster-gcp-creds
+  region: us-east1
 ```
 
-#### Create Cluster on GCP
-
-Credentials will be read from `~/.gcp/osServiceAccount.json`, this can be created by:
+### MachineConfig
 
- 1. Login to GCP console at https://console.cloud.google.com/
- 1. Create a service account with the owner role.
- 1. Create a key for the service account.
- 1. Select JSON for the key type.
- 1. Download resulting JSON file and save to `~/.gcp/osServiceAccount.json`.
+To manage `MachinePools` Day 2, you need to define these as well. The definition must match what was specified in the `InstallConfig`.
```

Comment:
The definition of the worker pool should mostly match what was specified in InstallConfig to prevent replacement of all worker nodes. (although there may be cases where you wish to apply slightly different worker config day 2)

 InstallConfig is limited to the one worker pool, but Hive can sync additional machine pools day 2.

Reply:
added this. Can you recheck it?

Reply:
Looks good, thanks!